### PR TITLE
Devise login

### PR DESCRIPTION
**Pull request by dev89s**
1. Removed `current_user` method from Application controller
2. Installed Devise Gem and installed in the project
3. Made new post and commenting and liking functionality requiring login for the user.
4. Also added Sign Up, Sign In and Log Out functionality to the main page
5. User logs in with a combination of email and password
6. Passwords are stored in the database as encrypted
7. registration requires email confirmation
8. There is a capability to reset password
9. Login view matches the wireframe layout

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <%= f.label :remember_me %>
+      </div>
+    <% end %>
+
+    <div class="actions">
+      <%= f.submit "Log in" %>
+    </div>
+  <% end %>
+  <%= render "devise/shared/links" %>
+</div>
```
Comment (danmainah) [inline app/views/devise/sessions/new.html.erb]: - [x] @dev89s and @akos29 Great work adding the login form, however, the design doesn't match the wireframe given, below is a comparison, with highlights on the differences.


Wireframe Design          |  Your design     
:-------------------------:|:-------------------------:
image  | image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb]: - [x] @dev89s and @akos29 Great work adding the Signup form, however, the design doesn't match the wireframe given, below is a comparison, with highlights on the differences.


Wireframe Design          |  Your design     
:-------------------------:|:-------------------------:
image  | !image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @dev89s and @akos29 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Install and setup the device correctly.
✔️  Removed current_user method in ApplicationController as devise provides it for us.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- @Team it would have been nice if you added instructions on how to run the app locally in your getting started section.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @dev89s, @akos29, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 Great job working on the login and signup pages to match the wireframe designs as requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**